lib/pkgr/builder.rb

```diff
@@ -284,17 +287,23 @@ def distribution
       @distribution ||= Distributions.current(config)
     end
 
-    # List of available buildpacks for the current distribution.
-    def buildpacks
-      distribution.buildpacks
-    end
-
-    # Buildpack detected for the app, if any.
-    def buildpack_for_app
+    # Buildpacks detected for the app, if any. If multiple buildpacks are explicitly specified, all are used
+    def buildpacks_for_app
       raise "#{source_dir} does not exist" unless File.directory?(source_dir)
-      @buildpack_for_app ||= buildpacks.find do |buildpack|
-        buildpack.setup(config.edge, config.home)
-        buildpack.detect(source_dir)
+      @buildpacks_for_app ||= begin
+        mode, buildpacks = distribution.buildpacks
+        case mode
+        when :custom
+          buildpacks.find_all do |buildpack|
+            buildpack.setup(config.edge, config.home)
+            buildpack.detect(source_dir)
+          end
+        else
+          [buildpacks.find do |buildpack|
+            buildpack.setup(config.edge, config.home)
+            buildpack.detect(source_dir)
+          end].compact
```

Comment:
Yes, but an app still might match multiple. Your proposal would still be a change in behaviour. Heroku and the multi-buildpack buildpack both require explicit specification of buildpacks when you want to use more than one.